src/main/scala/com/amazon/deequ/schema/ColumnName.scala

```diff
@@ -0,0 +1,109 @@
+/**
+  * Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+  *
+  * Licensed under the Apache License, Version 2.0 (the "License"). You may not
+  * use this file except in compliance with the License. A copy of the License
+  * is located at
+  *
+  *     http://aws.amazon.com/apache2.0/
+  *
+  * or in the "license" file accompanying this file. This file is distributed on
+  * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+  * express or implied. See the License for the specific language governing
+  * permissions and limitations under the License.
+  *
+  */
+
+package com.amazon.deequ.schema
+
+object ColumnName {
+
+  type Sanitized = Either[SanitizeError, String]
+
+  /**
+    * Sanitizes the input column name by ensuring that it is escaped with backticks.
+    *
+    * The resulting String is the escaped input column name, which is safe to use in
+    * any Spark SQL statement.
+    */
+  def sanitizeForSql(columnName: String): Sanitized =
+    if (columnName == null) {
+      Left(NullColumn)
+
+    } else {
+      val (prefix, suffix, insideColumnName) = {
+        val prefix = if (!columnName.startsWith("`")) "`" else ""
+        val suffix = if (!columnName.endsWith("`")) "`" else ""
+        val inside1 = if (prefix.isEmpty) {
+          columnName.slice(1, columnName.length)
+        } else {
+          columnName
+        }
+        val inside2 = if (suffix.isEmpty) {
+          inside1.slice(0, inside1.length - 1)
+        } else {
+          inside1
+        }
+        (prefix, suffix, inside2)
+      }
+
+      if (insideColumnName.contains("`")) {
+        Left(ColumnNameHasBackticks(columnName))
+      } else {
+        Right(s"$prefix$columnName$suffix")
+      }
+    }
+
+  /** Obtains the `String` value if `Right` or throws the `SanitizeError` if `Left`. */
+  def getOrThrow(x: Sanitized): String = x match {
+    case Left(e) => throw e
+    case Right(str) => str
+  }
+
+  /**
+    * Obtains the `String` pair if both are `Right`, otherwise throws the error(s).
+    *
+    * If only one of the sanitizations failed, then a `SanitizeError` type is thrown.
+    * If both fail, then an `IllegalArgumentException` is thrown and its message contains
+    * both of the `SanitizeError` messages.
+    */
+  def getOrThrow(x: (Sanitized, Sanitized)): (String, String) = x match {
+    case (Right(cA), Right(cB)) => (cA, cB)
+    case (Left(eA), Left(eB)) => throw new IllegalArgumentException(
+      s"Cannot sanitize two column names:\n$eA\n$eB"
+    )
+    case (Left(e), _) => throw e
+    case (_, Left(e)) => throw e
+  }
+
+  /** Alias for `sanitizeForSql | getOrThrow`. */
+  def sanitize(columnName: String): String =
+    getOrThrow(sanitizeForSql(columnName))
+
+  /** Inverse of `sanitize`: removes surrounding backticks, if present. */
+  def desanitize(maybeSanitizedName: String): String =
```

Comment:
why cant we use the replaceAll method with ` as argument?

Reply:
We could do that, especially since `maybeSanitizedName` _should_ only have backticks in the beginning and end, so removing all `\`` _should_ be equivalent in our use cases.

I did, however, write this function with the explicit intent of being an inverse of `sanitize`. I can def change the implementation to `_.removeAll("`")`. Given this intent, for the review, should I change?

Reply:
Re-wrote this awkward definition to slice into the string directly. I think this new definition is much more straightforward _and_ it is an inverse of `sanitize`:
```scala
maybeSanitizedName.slice(
    if(maybeSanitizedName.startsWith("`")) 1 else 0,
    maybeSanitizedName.length - {if(maybeSanitizedName.endsWith("`")) 1 else 0 }
)
```

Reply:
Unfortunately, I had to roll-back this change (broke tests :-/). I updated names to be consistent w/ the ones in `sanitizeForSql`.